### ISSUE TITLE
Discriminator support in JSON samples

### ISSUE THREAD

**Issue by ben-foster-cko** [Rebilly/ReDoc#459]
Are there plans, following the upgrade to 2.0 to support discriminated fields in JSON payloads? Currently these are not included in the examples.

**Comment by trip-somers** [comment]
I also have this question. This would be a great feature that would allow people to maintain their examples just by documenting/updating their requests and responses.

The current alternative/work-around is to explicitly define examples for each discriminator choice which is obviously cumbersome from a documentation maintenance perspective.

That said, I'm sure this isn't a small request. It would be nice to know if it's on the road map and/or when it might be prioritized. Given the current state of the examples in my documentation, I would likely be able to contribute to this on-the-clock. :-)

**Comment by trip-somers** [comment]
Digging into this a bit, it looks like it works if you define the request body as a `oneOf`, but not if a sub-object has a discriminator.

**Comment by jim5359** [comment]
For me, the models work perfectly for discriminator.  But the examples do not work if the sub-object has a discriminator.  See example attached.  The result is the same for both requests and responses.

![redoc-product-api](https://user-images.githubusercontent.com/12026228/43551085-91e8cf00-959a-11e8-8953-58f9a37825ec.png)

**Comment by cmeeren** [comment]
I too experience problems when using `oneOf` on a sub-object (with or without a discriminator doesn't matter).

**Comment by tjf** [comment]
@RomanGotsiy Any thoughts or updates on this?

**Comment by sayanroy7** [comment]
Anyone has any update or example of how to achieve this?

**Comment by atte-backman** [comment]
Yes, this is a bit annoying. Any plans going on tackling this one?

**Comment by toby-knott-cko** [comment]
@RomanHotsiy Has there been any progress on this?

**Comment by epark-brex** [comment]
Bump!
